library/index.c

```diff
@@ -323,24 +323,25 @@ int eblob_index_blocks_fill(struct eblob_base_ctl *bctl)
 }
 
 
-static struct eblob_disk_control *eblob_find_on_disk(struct eblob_backend *b,
-		struct eblob_base_ctl *bctl, struct eblob_disk_control *dc,
+static int eblob_find_on_disk(struct eblob_backend *b,
```

Comment:
In the worst possible scenario linear search could run away from within block boundaries (not over the whole file), but it requires that index repeatedly contains only the same key up to the end of the index file. This is not the case.

Your changes at 413+ lines were added to address this issue, weren't they?